homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -314,3 +404,66 @@ def native_value(self) -> datetime.datetime | float:
         assert envoy.data.inverters is not None
         inverter = envoy.data.inverters[self._serial_number]
         return self.entity_description.value_fn(inverter)
+
+
+class EnvoyEnchargeEntity(CoordinatorEntity[EnphaseUpdateCoordinator], SensorEntity):
+    """Envoy Encharge sensor entity."""
+
+    entity_description: EnvoyEnchargeSensorEntityDescription | EnvoyEnchargePowerSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: EnphaseUpdateCoordinator,
+        description: EnvoyEnchargeSensorEntityDescription
+        | EnvoyEnchargePowerSensorEntityDescription,
+        serial_number: str,
+    ) -> None:
+        """Initialize Encharge entity."""
+        self.entity_description = description
+        self._serial_number = serial_number
+        envoy_serial_num = coordinator.envoy.serial_number
+        assert envoy_serial_num is not None
+        self._attr_unique_id = f"{serial_number}_{description.key}"
+        assert coordinator.envoy.data is not None
+        assert coordinator.envoy.data.encharge_inventory is not None
```

Comment:
We can make some of properties that need an assert properties on the base class in a future PR

Than we can just do `self.data` and `self.envoy_serial_num`

Reply:
https://github.com/home-assistant/core/pull/98012